tests/harvester/test_utils.py

```diff
@@ -10,3 +12,18 @@ def test_binding_id_from_dc(mets_dc_identifier):
     Test that DC identifiers are parsed to binding IDs correctly.
     """
     assert utils.binding_id_from_dc(mets_dc_identifier) == "379973"
+
+
+def test_make_intermediate_dirs(sftp_client, mocker):
+    """
+    Test that intermediate directories on a remote host are created as expected.
+    TODO Testing that the function works as expected after OSError.
```

Comment:
If we leave this TODO here without doing anything about it, I'm pretty certain we'll never do anything about it. As this is about a branch whose inner workings are somewhat complicated, I suggest we figure a way to test it instead of leaving it untested as we might with something really trivial but hard to test.

Reply:
Fixed.